rust/lance-index/src/vector/v3/storage.rs

```diff
@@ -49,4 +50,8 @@ pub trait VectorStore: Send + Sync {
     fn dist_calculator_from_id(&self, id: u32) -> Self::DistanceCalculator<'_>;
 
     fn distance_between(&self, a: u32, b: u32) -> f32;
+
+    fn dist_calculator_from_native<T: Num>(&self, _query: &[T]) -> Self::DistanceCalculator<'_> {
```

Comment:
why do we provide default implementation yet throw `todo!()` here?

Reply:
I don't know of an easy way to translate `T:Num` into `ArrayRef` that's why this is todo.

Reply:
yeah, let's keep it like this for now i guess. I'll clean up the storage and fix this tomorrow